pkg/promtail/scrapeconfig/scrapeconfig.go

```diff
@@ -2,26 +2,34 @@ package scrapeconfig
 
 import (
 	"fmt"
+	"reflect"
 	"time"
 
 	"github.com/prometheus/common/model"
-	"github.com/weaveworks/common/server"
-
 	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/file"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
 	"github.com/prometheus/prometheus/pkg/relabel"
+	"github.com/weaveworks/common/server"
 
 	"github.com/grafana/loki/pkg/logentry/stages"
 )
 
 // Config describes a job to scrape.
 type Config struct {
-	JobName        string                `yaml:"job_name,omitempty"`
-	PipelineStages stages.PipelineStages `yaml:"pipeline_stages,omitempty"`
-	JournalConfig  *JournalTargetConfig  `yaml:"journal,omitempty"`
-	SyslogConfig   *SyslogTargetConfig   `yaml:"syslog,omitempty"`
-	PushConfig     *PushTargetConfig     `yaml:"loki_push_api,omitempty"`
-	RelabelConfigs []*relabel.Config     `yaml:"relabel_configs,omitempty"`
-	discovery.Config
+	JobName                string                 `yaml:"job_name,omitempty"`
+	PipelineStages         stages.PipelineStages  `yaml:"pipeline_stages,omitempty"`
+	JournalConfig          *JournalTargetConfig   `yaml:"journal,omitempty"`
+	SyslogConfig           *SyslogTargetConfig    `yaml:"syslog,omitempty"`
+	PushConfig             *PushTargetConfig      `yaml:"loki_push_api,omitempty"`
+	RelabelConfigs         []*relabel.Config      `yaml:"relabel_configs,omitempty"`
+	ServiceDiscoveryConfig ServiceDiscoveryConfig `yaml:",inline"`
+}
+
+type ServiceDiscoveryConfig struct {
+	KubernetesSDConfigs []*kubernetes.SDConfig `yaml:"kubernetes_sd_configs"`
+	StaticConfigs       discovery.StaticConfig `yaml:"static_configs"`
+	FileSdConfigs       []*file.SDConfig       `yaml:"file_sd_configs,omitempty"`
```

Comment:
i think there are other types we need to have here right? I know people use ec2 scrape configs as an example